mlflow/pytorch/torchserve/deploy/serve.py

```diff
@@ -0,0 +1,264 @@
+
+    def validate_mandatory_arguments(self):
+        """
+        Validate the mandatory arguments is present if not raise exception
+        """
+
+        if not self.server_config["version"]:
+            raise Exception("Environment Variable VERSION - missing")
+
+        if not self.server_config["model_file"]:
+            raise Exception("Environment Variable MODEL_FILE - missing")
+
+        if not self.server_config["handler_file"]:
+            raise Exception("Environment Variable HANDLER_FILE - missing")
+
+    def create_deployment(self, name, model_uri, flavor=None, config=None):
+        """
+        Deploy the model at the model_uri to the specified target
+        """
+        self.validate_mandatory_arguments()
+        mar_file_path = self.generate_mar_file(
+            model_name=name,
+            version=self.server_config["version"],
+            model_file=self.server_config["model_file"],
+            handler_file=self.server_config["handler_file"],
+            extra_files=self.server_config["extra_files"],
+            model_uri=model_uri,
+        )
+
+        self.register_model(
+            mar_file_path=mar_file_path,
+            name=name,
+            model_uri=model_uri,
+            flavor=flavor,
+            config=config,
+        )
+        return {"name": name, "flavor": flavor}
+
+    def delete_deployment(self, name):
+        """
+        Delete the deployment with the name given at --name from the specified target
+        """
+
+        version = self.server_config["version"]
+        url = "{}/{}/{}/{}".format(self.management_api, "models", name, version)
+        resp = requests.delete(url)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return None
+
+    def update_deployment(self, name, model_uri=None, flavor=None, config=None):
+        """
+        Update the deployment with the name given at --name from the specified target
+
+        Using -C or --config additional parameters shall be updated for the corresponding model
+        """
+
+        query_path = ""
+
+        if config is not None:
+            for key in config:
+                query_path += "&" + key + "=" + str(config[key])
+
+            query_path = query_path[1:]
+
+        url = "{}/{}/{}?{}".format(self.management_api, "models", name, query_path)
+        resp = requests.put(url)
+
+        if resp.status_code != 202:
+            raise Exception("Unable to list deployments")
+        return {"flavor": flavor}
+
+    def list_deployments(self):
+        """
+        List the names of all model deployments in the specified target. These names can be used with
+        delete , update and get commands
+        """
+        url = "{}/{}".format(self.management_api, "models")
+        resp = requests.get(url)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return [resp.text]
```

Comment:
We actually expect this method to return a list of dicts where each dict contains a `name` key that can be used with the delete, update, and get commands. Currently, the `list_deployments` API doesn't support pagination, so we may have to loop over paginated models registered in TorchServe (i.e. make multiple requests to the [TorchServe `models` endpoint](https://github.com/pytorch/serve/blob/master/docs/management_api.md#list-models) and concatenate the results together). Note that to conform with the deploy plugin interface, we may also want to duplicate the `modelName` key returned by TorchServe as a `name` key in the returned dictionaries. Let me know if that makes sense!

Reply:
We have updated the list deployment method to query based on pagination. As suggested we have added the name of the deployment as key to the dictionary and output looks like this

[
'test': {'modelName': 'test', 'modelUrl': '/home/ubuntu/test.mar'},
'test1': {'modelName': 'test1', 'modelUrl': '/home/ubuntu/test1.mar'},
]

Note: https://github.com/pytorch/serve/blob/master/docs/management_api.md#list-models - Limit and the next page token are part of the url itself. if we need to implement user defined values for list deployment we may need to add the parameters in base class. 

For example `plugin.list_deployments(limit=2, next_page_token=2)` - To achieve this we need to add parameters in the base class.

Please let us know your comments on this.